### PR TITLE
fix(steps): add unit test and fix null cols for impute

### LINKED ISSUE #117: bug(steps): handle col with all nulls in impute
For all the impute, we need to handle cols with all nulls. We need to tell the user the column is all nulls by failing the imputing or throw warnings. 

I prefer `fail` the impute. 

```
import ibis
import ibis_ml as ml
import numpy as np

a = ibis.memtable(
    {"all_null": np.array([np.nan, np.nan, np.nan], dtype="float64")}
)
step = ml.ImputeMean(ml.numeric())
step.fit_table(a, ml.core.Metadata())
step.transform_table(a)
```
```
┏━━━━━━━━━━┓
┃ all_null ┃
┡━━━━━━━━━━┩
│ float64  │
├──────────┤
│      nan │
│      nan │
│      nan │
└──────────┘
```

### PR DESCRIPTION
**Pull request by jitingxu1**
changes:

1. add unit test for _impute.py
2. raise a value error if one column is all NULL, because it does not fulfill users' original goal.

resolve #117

### CODE REVIEW

File: tests/test_impute.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+def test_fillna(train_table):
+    step = ml.FillNA("floating_col", 0)
+    step.fit_table(train_table, ml.core.Metadata())
+    assert step.is_fitted()
+    test_table = ibis.memtable({"floating_col": [None]})
+    result = step.transform_table(test_table)
+    expected = pd.DataFrame({"floating_col": [0]})
+    tm.assert_frame_equal(result.execute(), expected, check_dtype=False)
+
+    # test _fillna with None
```
Comment (deepyaman) [inline tests/test_impute.py:53]: Can you make this a second test, and parametrize with `None` and `NaN` inputs?

File: ibis_ml/steps/_impute.py [hunk 2]
```diff
@@ -14,6 +15,8 @@
 
 
 def _fillna(col, val):
+    if val is None or col.type().is_floating() and math.isnan(val):
```
Comment (deepyaman) [inline ibis_ml/steps/_impute.py:18]:
```suggestion
    if val is None or and math.isnan(val):
```
Why is the `is_floating()` check necessary?

Also, I think it would make sense to have separate error for `None` and `NaN` (i.e. be specific in the message, rather than saying it could be either)

**Review by deepyaman — CHANGES_REQUESTED** [review]
Left some comments, but maybe more fundamentally—why is this necessary/what motivated it? If somebody wants to fill their missing values with `None`, should I care? E.g. maybe they want to represent all the missing value as NULL (and not NaN), so they do this. Alternatively if you see scikit-learn `SimpleImputer`, they later let the user customize what is considered "missing".

**Comment by deepyaman** [comment]
> Left some comments, but maybe more fundamentally—why is this necessary/what motivated it? If somebody wants to fill their missing values with `None`, should I care? E.g. maybe they want to represent all the missing value as NULL (and not NaN), so they do this. Alternatively if you see scikit-learn `SimpleImputer`, they later let the user customize what is considered "missing".

I'm pushing to deprioritize this beyond today, unless you feel it's particularly pressing. Let's focus on getting the other PRs in for release.

File: tests/test_impute.py (hunk 1 above)
Comment (jitingxu1) [reply tests/test_impute.py]: done

File: ibis_ml/steps/_impute.py (hunk 2 above)
Comment (jitingxu1) [reply ibis_ml/steps/_impute.py]: It could be a string col, the `math.isnan(val)` will throw an error

**Comment by jitingxu1** [comment]
> Left some comments, but maybe more fundamentally—why is this necessary/what motivated it? If somebody wants to fill their missing values with `None`, should I care? E.g. maybe they want to represent all the missing value as NULL (and not NaN), so they do this. Alternatively if you see scikit-learn `SimpleImputer`, they later let the user customize what is considered "missing".

Instead of raise an error, I change it to throw a warning to the user.
